GymBuddies/ViewControllers/CreateExerciseViewController.m

```diff
@@ -0,0 +1,219 @@
+    cell.backgroundColor = [UIColor secondarySystemBackgroundColor];
+    NSLog(@"%@", self.bodyZones[indexPath.row][@"title"]);
+    self.exerciseBodyZoneTag = self.bodyZones[indexPath.row];
+}
+
+- (void)collectionView:(UICollectionView *)collectionView didDeselectItemAtIndexPath:(NSIndexPath *)indexPath{
+    UICollectionViewCell *cell = [self.bodyZoneCollectionView cellForItemAtIndexPath:indexPath];
+    cell.backgroundColor = [UIColor systemBackgroundColor];
+}
+
+
+#pragma mark -Selecting media
+
+
+- (IBAction)uploadVideo:(id)sender {
+    if([UIImagePickerController isSourceTypeAvailable:UIImagePickerControllerSourceTypePhotoLibrary]){
+        self.mediaPicker.sourceType = UIImagePickerControllerSourceTypePhotoLibrary;
+        self.mediaPicker.mediaTypes = @[(NSString*)kUTTypeMovie, (NSString*)kUTTypeAVIMovie, (NSString*)kUTTypeVideo, (NSString*)kUTTypeMPEG4];
+        [self presentViewController:self.mediaPicker animated:YES completion:nil];
+    }
+}
+
+
+- (IBAction)uploadImage:(id)sender {
+    if([UIImagePickerController isSourceTypeAvailable:UIImagePickerControllerSourceTypePhotoLibrary]){
+        self.mediaPicker.sourceType = UIImagePickerControllerSourceTypePhotoLibrary;
+        self.mediaPicker.mediaTypes = @[(NSString*)kUTTypeImage];
+        [self presentViewController:self.mediaPicker animated:YES completion:nil];
+    }
+}
+
+
+
+- (void)imagePickerController:(UIImagePickerController *)picker didFinishPickingMediaWithInfo:(NSDictionary<UIImagePickerControllerInfoKey,id> *)info{
+    NSString *mediaType = [info objectForKey:UIImagePickerControllerMediaType];
+    if([mediaType isEqualToString:(NSString*)kUTTypeMovie] ||  [mediaType isEqualToString:(NSString*)kUTTypeAVIMovie] || [mediaType isEqualToString:(NSString*)kUTTypeVideo] || [mediaType isEqualToString:(NSString*)kUTTypeMPEG4]){
+        NSURL *urlVideo = [info objectForKey:UIImagePickerControllerMediaURL];
+        PFFileObject *video = [ParseAPIManager getPFFileFromURL:urlVideo];
+        self.exerciseVideo = video;
+    } else {
+        self.imagePreview.image = [info objectForKey:UIImagePickerControllerOriginalImage];
+        PFFileObject *image = [ParseAPIManager getPFFileFromImage:self.imagePreview.image];
+        self.exerciseImage = image;
+        [self dismissViewControllerAnimated:YES completion:nil];
+    }
+    /*self.imagePreview.image = [info objectForKey:UIImagePickerControllerOriginalImage];
+    [self dismissViewControllerAnimated:YES completion:nil];*/
+}
+
+
+#pragma mark -Alerts
+
+-(void)_failedFetchingAlert:(NSString *)message{
+    UIAlertController *alert = [UIAlertController alertControllerWithTitle:@"Error fetching data" message:message preferredStyle:UIAlertControllerStyleAlert];
+    
+    UIAlertAction *okAction = [UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleDefault handler:nil];
+    [alert addAction:okAction];
+    
+    [self presentViewController:alert animated:YES completion:nil];
```

Comment:
how about extracting all this into a helper and just passing in the differences (seems just the title)

Reply:
Ok. I'll try that. Maybe I'll change the name of the CommonValidations.h to include the alert methods